File: TwoPointer/twosum.cpp
#include <iostream>
#include <vector>

std::vector<int> twoSum(std::vector<int> &nums, int target) {
    int left = 0, right = nums.size() - 1;

    while (left < right) {
        int current_sum = nums[left] + nums[right];
        if(current_sum == target) {
            return {left, right};
        } else if (current_sum < target) {
            left++;
        } else {
            right--;
        }
    }
    return {};
}

int main() {
    std::vector<int> nums = {1, 2, 3, 4, 5, 6};
    std::vector<int> result = twoSum(nums, 6);
    if (!result.empty()){
        std::cout << "Indicies: " << result[0] << ", " << std::endl;
    }
    return 0;    
}

File: TwoPointer/two_sum.py
def two_sum(nums, target):
    left, right = 0, len(nums) - 1
    
    while left < right:
        current_sum = nums[left] + nums[right]
        if current_sum == target:
            return [left, right]
        elif current_sum < target:
            left += 1
        else:
            right -= 1
    
    return None

print(two_sum([1, 2, 3, 4, 5, 6], 6)) # Output: [1, 3]